lib/pure/sugar.nim

```diff
@@ -160,16 +156,12 @@ macro dump*(x: untyped): untyped =
   ## It accepts any expression and prints a textual representation
   ## of the tree representing the expression - as it would appear in
   ## source code - together with the value of the expression.
-  ##
-  ## As an example,
-  ##
-  ## .. code-block:: nim
-  ##   let
-  ##     x = 10
-  ##     y = 20
-  ##   dump(x + y)
-  ##
-  ## will print ``x + y = 30``.
+  runnableExamples:
+    let
+      x = 10
+      y = 20
+    dump(x + y) # will print `x + y = 30`
```

Comment:
please do not print strings here.

Reply:
in future PR we should deprecate `dump` in favor of a proc that returns a string; more generally, API's should almost never `echo`, that should be left up to user code (it's always less flexible, can't as easily be unit-tested, etc)

Reply:
Returning a string produces evil allocations though... And `dup` on a `var string` is too inconvenient for `dump`. *shrug*

Reply:
> Returning a string produces evil allocations though

I don't see how that's a problem for a proc that's meant for debugging templates/macros

> And dup on a var string is too inconvenient for dump. shrug

no need for dup or var string. see https://github.com/nim-lang/Nim/pull/16841